### PR TITLE
MM-25261: Adds archived channels to admin console channel list view.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary

Displays archived channels in the channels list view. Add ability to archive and unarchive channels from their details view. 

Also this PR reorders the channel moderation widget among the other widgets in the channel details view.

#### Ticket Link

* https://mattermost.atlassian.net/browse/MM-25261
* https://mattermost.atlassian.net/browse/MM-25266

#### Related Pull Requests

- Has server changes: https://github.com/mattermost/mattermost-server/pull/14721
- Has redux changes: https://github.com/mattermost/mattermost-redux/pull/1168

#### Screenshots

localhost_8065_admin_console_user_management_channels_5jkagupgaiba7xs5z7dauampbc

localhost_8065_admin_console_user_management_channels_5jkagupgaiba7xs5z7dauampbc

 
localhost_8065_admin_console_user_management_channels_5jkagupgaiba7xs5z7dauampbc (2)

### CODE REVIEW

File: components/admin_console/team_channel_settings/channel/details/channel_profile.tsx [hunk 1]
```diff
@@ -26,7 +44,7 @@ export const ChannelProfile: React.SFC<ChannelProfileProps> = (props: ChannelPro
             subtitleId={t('admin.channel_settings.channel_detail.profileDescription')}
             subtitleDefault='Summary of the channel, including the channel name.'
         >
-            <div className='group-teams-and-channels'>
+            <div className='group-teams-and-channels AdminChannelDetails'>
```
Comment (fmunshi) [inline components/admin_console/team_channel_settings/channel/details/channel_profile.tsx:47]: do we need to keep the old class?

**Comment by fmunshi** [comment]
Also looks like you've got some conflicts to take care of @mkraft

**Comment by fmunshi** [comment]
Paging @catalintomai for review

File: components/admin_console/team_channel_settings/channel/details/channel_profile.tsx (hunk 1 above)
Comment (mkraft) [reply components/admin_console/team_channel_settings/channel/details/channel_profile.tsx:47]: I'm not sure how to deal with this. My thinking was to use the new naming convention, but there are still a bunch of styles tied to `.group-teams-and-channels`. What do you think? We could:

a) replace `AdminChannelDetails` with `group-teams-and-channels` and tie the new styles to it instead
c) update `group-teams-and-channels` to be `GroupTeamsAndChannels` and then all of the associated states like `group-teams-and-channels--body`, etc, this does mean updating a bunch of things including tests, not insurmountable just noting it
d) new stuff goes under `AdminChannelDetails` (ie leave it as-is in the PR)

Thoughts? Other ideas?

**Comment by thefactremains** [comment]
adding a note to remind myself to update this docs page before release: https://docs.mattermost.com/help/getting-started/organizing-conversations.html#archiving-a-channel

**Pull request merged**